### PR TITLE
fix(sqlite): parse date error

### PR DESCRIPTION
**Pull request by skypesky**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

- [x] fix: #16340 

## Todos

- [x] null

### CODE REVIEW

File: test/unit/dialects/sqlite/data-types.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const chai = require('chai'),
+  expect = chai.expect,
+  Support = require('../../support'),
+  dialect = Support.getTestDialect(),
+  BaseTypes = require('sequelize/lib/data-types'),
+  DataTypes = require('sequelize/lib/dialects/sqlite/data-types')(BaseTypes);
+
+if (dialect.match(/^sqlite/)) {
+  describe('[SQLITE Specific] DataTypes', () => {
```
Comment (WikiRik) [inline test/unit/dialects/sqlite/data-types.test.js:11]: Are these tests really sqlite specific? Is the behaviour different on other dialects?

Comment (skypesky) [reply test/unit/dialects/sqlite/data-types.test.js:11]: Ok,Here's what I think。

1. I don't know the details of how DATE is handled in other dialects, but it appears from the source code that different dialects handle dates differently from each other.
2. the problem I'm having only occurs in sqlite, so I'm just fixing the parse function in sqlite and making sure it's working.

**Comment by skypesky** [comment]
Can this pr be merged?
This test is probably wrong because of instability, and this change has nothing to do with postgress.

File: test/unit/dialects/sqlite/data-types.test.js (hunk 1 above)
Comment (WikiRik) [reply test/unit/dialects/sqlite/data-types.test.js:11]: I think the other dialects should also handle these test cases the same way, so I'd like to see these tests running on all dialects and therefore be added to an existing test suite

**Comment by ephys** [comment]
I'm not convinced we should do this. Our Date DataType inserts dates in the ISO format, not as numbers. If the goal is to fix #16340, it's `queryInterface.insert` that should be fixed instead.

Once that is fixed, any Date persisted to the database as a number will not be caused by us, therefore we can't know what level of precision it is meant to represent. It could be milliseconds, nanoseconds, seconds.

You can always extend DATE in your project to suit your needs but I don't think we should apply this to all users

**Comment by skypesky** [comment]
> I'm not convinced we should do this. Our Date DataType inserts dates in the ISO format, not as numbers. If the goal is to fix #16340, it's `queryInterface.insert` that should be fixed instead.
> 
> Once that is fixed, any Date persisted to the database as a number will not be caused by us, therefore we can't know what level of precision it is meant to represent. It could be milliseconds, nanoseconds, seconds.
> 
> You can always extend DATE in your project to suit your needs but I don't think we should apply this to all users

`queryInterface.insert` is just an example of the error it throws, even just querying for records will throw this error, I get this error querying for data when migrating prisma to sequelize (e.g. User.findAll(), User.find()). Just fixing it at `queryInterface.insert` is not enough.

**Comment by ephys** [comment]
Yes but we're not responsible for the format prisma used

**Pull request closed without merging**